synapse/handlers/federation_event.py

```diff
@@ -542,8 +542,24 @@ async def update_state_for_partial_state_event(
                 destination, event
             )
 
+            # There are three possible cases for (state_ids, partial_state):
+            #   * `state_ids` and `partial_state` are both `None` if we had all the
+            #     prev_events. The prev_events may or may not have partial state and
+            #     we won't know until we compute the event context.
+            #   * `state_ids` is not `None` and `partial_state` is `False` if we were
+            #     missing any prev_events. We calculated the full state after the
+            #     prev_events.
```

Comment:
Probably not worth worrying too much about this if you're planning to refactor `_get_state_after_missing_prevs`, but:

```suggestion
            #   * `state_ids` is not `None` and `partial_state` is `False` if we were
            #     missing some prev_events (but we have full state for any we did
            #     have). We calculated the full state after the prev_events.
```